tracer/src/Datadog.Trace/ContinuousProfiler/ContextTracker.cs

```diff
@@ -0,0 +1,120 @@
+// <copyright file="ContextTracker.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ContinuousProfiler
+{
+    internal class ContextTracker
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ContextTracker));
+
+        private readonly ProfilerStatus _status;
+        private readonly bool _isFlagSet;
+        private readonly ThreadLocal<IntPtr> _traceContextPtr;
+
+        public ContextTracker(ProfilerStatus status)
+        {
+            _status = status;
+            _isFlagSet = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.CodeHotspotsEnabled)?.ToBoolean() ?? false;
+            _traceContextPtr = new ThreadLocal<IntPtr>();
+            Log.Information("CodeHotspots feature is {IsEnabled}.", _isFlagSet ? "enabled" : "disabled");
+        }
+
+        public bool IsEnabled
+        {
+            get
+            {
+                return _status.IsProfilerReady && _isFlagSet;
+            }
+        }
+
+        public void Set(ulong localRootSpanId, ulong spanId)
+        {
+            WriteToNative(new SpanContext(localRootSpanId, spanId));
+        }
+
+        public void Reset()
+        {
+            WriteToNative(SpanContext.Zero);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Methods that actually do calls to native must not be inlined.
So in case of partial-trust environment, the parent calls those methods with a try/catch and we can catch the exception and move along.
If the call is inlined, the exception won't be caught

Reply:
Cool thanks.

> So in case of partial-trust environment

BTW, we don't support partial trust anymore.